File: src/Core/Logistics.Domain/Entities/Payment.cs
using Logistics.Domain.Core;
using Logistics.Domain.ValueObjects;
using Logistics.Shared.Enums;

namespace Logistics.Domain.Entities;

public class Payment : Entity, ITenantEntity
{
    public decimal Amount { get; set; }
    public PaymentStatus Status { get; set; }
    public PaymentMethod? Method { get; set; }
    public PaymentFor PaymentFor { get; set; }
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
    public DateTime? PaymentDate { get; set; }
    public string? Comment { get; set; }
    public Address BillingAddress { get; set; } = Address.NullAddress;

    public void SetStatus(PaymentStatus status)
    {
        if (status == PaymentStatus.Paid)
        {
            PaymentDate = DateTime.UtcNow;
        }

        Status = status;
    }
}

File: src/Client/Logistics.OfficeApp/src/app/core/models/payment/payment.ts
import {PaymentFor, PaymentMethod, PaymentStatus} from '@core/enums';


export interface Payment {
  id: string;
  createdDate: string;
  paymentDate?: string;
  method?: PaymentMethod;
  amount: number;
  status: PaymentStatus;
  paymentFor: PaymentFor;
  comment?: string;
}
